### PR TITLE
"pygmt.Figure.grdimage": Expand docstring for "nan_transparent" parameter

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to expand the docstring for the `nan_transparent` parameter of [`pygmt.Figure.grdimage`](https://www.pygmt.org/dev/api/generated/pygmt.Figure.grdimage.html#pygmt.Figure.grdimage), based on the [upstream GMT documentation for **Q**](https://docs.generic-mapping-tools.org/dev/grdimage.html#q).

Somehow / Maybe related to this forum post: https://forum.generic-mapping-tools.org/t/how-to-not-plot-value-zero-on-grid/3783/5

**Preview** at https://pygmt-dev--2461.org.readthedocs.build/en/2461/api/generated/pygmt.Figure.grdimage.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/grdimage.py [hunk 1]
```diff
@@ -148,10 +148,14 @@ def grdimage(self, grid, **kwargs):
     no_clip : bool
         Do **not** clip the image at the frame boundaries (only relevant
         for non-rectangular maps) [Default is ``False``].
-    nan_transparent : bool
+    nan_transparent : bool or str
+        [**+z**\ *value*][*color*]
         Make grid nodes with z = NaN transparent, using the color-masking
```
Comment (seisman) [inline pygmt/src/grdimage.py:153]: > using the color-masking feature in PostScript Level 3 (the PS device must support PS Level 3).

PostScript Level 3 was released in 1997. Is it safe to assume that all printers support PS Level 3 now? Or do we care about PS Level 3 if we always convert PS to PDF then print?

**Review by michaelgrund — APPROVED** [review]

File: pygmt/src/grdimage.py (hunk 1 above)
Comment (seisman) [reply pygmt/src/grdimage.py:153]: Ping @PaulWessel just in case he missed this comment.

**Review by maxrjones — APPROVED** [review]
Nice!

File: pygmt/src/grdimage.py (hunk 1 above)
Comment (maxrjones) [reply pygmt/src/grdimage.py:153]: I think it's fine to merge this keeping the warning

**Pull request merged**